pkg/store/bucket_test.go

```diff
@@ -16,6 +16,149 @@ import (
 	"github.com/prometheus/tsdb/labels"
 )
 
+// TestBucketBlockSet with blocks which have the same time range
+// but different resolutions.
+func TestBucketBlockSet_Duplicated(t *testing.T) {
+	defer leaktest.CheckTimeout(t, 10*time.Second)()
+
+	set := newBucketBlockSet(labels.Labels{})
+
+	type resBlock struct {
+		mint, maxt int64
+		window     int64
+	}
+	input := []resBlock{
```

Comment:
What do you think about joinging those together in form of table test? essentiall add those as test cases to `TestBucketBlockSet_addGet`?  might be more consistent and easier to read just "cases" as test itself feels similar TBH?

Reply:
These ones are a bit special because we are doing a property-based tested instead of an ad-hoc one. I have removed the other test cases that I've added before and only left this one in - this already covers the "interleaved resolutions" case.

Reply:
:+1: